### PR TITLE
[Telemetry Operator] Bump kubebuilder and controller-runtime dependencies

### PR DESCRIPTION
**Pull request by dennis-ge**
**Description**

Changes proposed in this pull request:

- Bump kubebuilder to 3.6
- Update README.md to align to current status of operator
- Bump controller runtime to 0.12.3, fixing #15117 

**Related issue(s)**
Relates to #15117

### CODE REVIEW

**Review by chrkl — DISMISSED** [review]

File: components/telemetry-operator/README.md [hunk 1]
```diff
@@ -2,83 +2,22 @@
 
 ## Overview
 
-The telemetry operator contains a logging controller that generates a Fluent Bit configuration from one or more LogPipeline custom resources. The controller ensures that all Fluent Bit Pods run the current configuration by deleting Pods after the configuration has changed. See all [CRD attributes](apis/telemetry/v1alpha1/logpipeline_types.go) and an [example](config/samples/telemetry_v1alpha1_logpipeline.yaml).
+The telemetry operator contains a logging controller that generates a Fluent Bit configuration from one or more LogPipeline and LogParser Custom Resources. The controller ensures that all Fluent Bit Pods run the current configuration by deleting Pods after the configuration has changed. See all [CRD attributes](apis/telemetry/v1alpha1/logpipeline_types.go) and some [examples](config/samples).
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:5]:
```suggestion
The telemetry operator contains a logging controller that generates a Fluent Bit configuration from one or more LogPipeline and LogParser custom resources. The controller ensures that all Fluent Bit Pods run the current configuration by deleting Pods after the configuration has changed. See all [CRD attributes](apis/telemetry/v1alpha1/logpipeline_types.go) and some [examples](config/samples).
```
https://kyma-project.io/community/guidelines/content-guidelines/02-style-and-terminology#capitalization

File: components/telemetry-operator/README.md [hunk 2]
```diff
@@ -2,83 +2,22 @@
...
-   ```
+The operator has been bootstrapped with [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) 3.6.0. Additional APIs can also be [added by Kubebuilder](https://book.kubebuilder.io/cronjob-tutorial/new-api.html).
 
 ## Development
 
 ### Prerequisites
-- Install [kubebuilder 3.2.0](https://github.com/kubernetes-sigs/kubebuilder) which is the base framework for this controller
-- Install [kustomize](https://github.com/kubernetes-sigs/kustomize) which lets you customize raw, template-free `yaml` files during local development
-- Install [Golang 1.18](https://golang.org/dl/) or newer (for local execution)
+- Install [kubebuilder 3.6.0](https://github.com/kubernetes-sigs/kubebuilder) which is the base framework for this controller
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:18]:
```suggestion
- Install [kubebuilder 3.6.0](https://github.com/kubernetes-sigs/kubebuilder), which is the base framework for this controller.
```

File: components/telemetry-operator/README.md [hunk 3]
```diff
@@ -2,83 +2,22 @@
...
+- Install [kustomize](https://github.com/kubernetes-sigs/kustomize) which lets you customize raw, template-free `yaml` files during local development (see `kustomize` make target)
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:19]:
```suggestion
- Install [kustomize](https://github.com/kubernetes-sigs/kustomize), which lets you customize raw, template-free YAML files during local development (see `kustomize` make target).
```

What does "(see `kustomize` make target)" mean??

File: components/telemetry-operator/README.md [hunk 4]
```diff
@@ -2,83 +2,22 @@
...
+- Install [Golang 1.19](https://golang.org/dl/) or newer (for local execution)
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:20]:
```suggestion
- Install [Golang 1.19](https://golang.org/dl/) or newer (for local execution).
```

File: components/telemetry-operator/README.md [hunk 5]
```diff
@@ -2,83 +2,22 @@
...
 - Install [Docker](https://www.docker.com/get-started)
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:21]:
```suggestion
- Install [Docker](https://www.docker.com/get-started).
```

File: components/telemetry-operator/README.md [hunk 6]
```diff
@@ -2,83 +2,22 @@
...
 - Install [OpenSSL](https://www.openssl.org/) to generate webhook certificate for local execution
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:22]:
```suggestion
- Install [OpenSSL](https://www.openssl.org/) to generate a webhook certificate for local execution.
```

File: components/telemetry-operator/README.md (hunk 3 above)
Comment (dennis-ge) [reply components/telemetry-operator/README.md:19]: we have a dedicated make target to install kustomize. So executing `make kustomize` installs kustomize on the fly

File: components/telemetry-operator/README.md [hunk 7]
```diff
@@ -2,85 +2,24 @@
...
 ## Development
 
 ### Prerequisites
-- Install [kubebuilder 3.2.0](https://github.com/kubernetes-sigs/kubebuilder) which is the base framework for this controller
-- Install [kustomize](https://github.com/kubernetes-sigs/kustomize) which lets you customize raw, template-free `yaml` files during local development
-- Install [Golang 1.18](https://golang.org/dl/) or newer (for local execution)
-- Install [Docker](https://www.docker.com/get-started)
-- Install [OpenSSL](https://www.openssl.org/) to generate webhook certificate for local execution
+- Install [kubebuilder 3.6.0](https://github.com/kubernetes-sigs/kubebuilder), which is the base framework for this controller.
+- Install [kustomize](https://github.com/kubernetes-sigs/kustomize) which lets you customize raw, template-free `yaml` files during local development (see `kustomize` make target)
```
Comment (dennis-ge) [inline components/telemetry-operator/README.md:19]:
```suggestion
- Install [kustomize](https://github.com/kubernetes-sigs/kustomize) which lets you customize raw, template-free YAML files during local development.
```

**Review by NHingerl — APPROVED** [review]

**Review by chrkl — APPROVED** [review]

**Pull request merged**